pkg/scheduler/scheduler.go

```diff
@@ -423,11 +420,15 @@ func (e entryOrdering) Less(i, j int) bool {
 	return a.Obj.CreationTimestamp.Before(&b.Obj.CreationTimestamp)
 }
 
-func (s *Scheduler) requeueAndUpdate(log logr.Logger, ctx context.Context, w *workload.Info, message string) {
-	added := s.queues.RequeueWorkload(ctx, w)
-	log.V(2).Info("Workload re-queued", "workload", klog.KObj(w.Obj), "queue", klog.KRef(w.Obj.Namespace, w.Obj.Spec.QueueName), "added", added)
-	err := workload.UpdateStatus(ctx, s.client, w.Obj, kueue.WorkloadAdmitted, corev1.ConditionFalse, "Pending", message)
-	if err != nil {
-		log.Error(err, "Could not update Workload status")
+func (s *Scheduler) requeueAndUpdate(log logr.Logger, ctx context.Context, e entry, message string) {
+	added := s.queues.RequeueWorkload(ctx, &e.Info, e.status != skipped)
```

Comment:
my brain is hurting again thinking about the double negative.

I think we should rename the variable to `immediate` (for immediate requeue).

then `immediate = e.status == skipped` reads pretty well.

Reply:
at this point `nominated` means inadmissible, so this check could be `e.status == nominated`; but renaming the `inadmissible` flag to `immediate` sgtm as well.

Reply:
`e.status == nominated` is ok here.
so we decide to rename `inadmissible` to `immediate`?

Reply:
I'm fine with either after Abdullah's comment. Your call, Alex :)

Reply:
actually no, `e.status == nominated` isn't the only inadmissible case, it is also when the status is not set at all! So lets do aldo's suggestion.

Reply:
Also, if the spec update fails in the API, we should also make immediate requeue. So it should be:

`immediate = (e.status == skipped || e.status == assumed)`

Btw, the only way it stays nominated is when it fails to be assumed. This happens if the workload was already assumed or the clusterqueue is gone. In which case it doesn't matter which type of requeue we do.

Reply:
2nd BTW: given all these problems trying to reason about these states, we should probably unit test this logic :)